subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,424 @@
+# -*- coding: utf-8 -*-
+import os
+import logging
+import re
+import json
+from zipfile import ZipFile, is_zipfile
+from rarfile import RarFile, is_rarfile
+from tempfile import NamedTemporaryFile
+
+from babelfish import Language, language_converters
+from requests import Session
+
+from . import ParserBeautifulSoup, Provider, get_version
+from .. import __version__
+from ..cache import SHOW_EXPIRATION_TIME, region
+from ..exceptions import AuthenticationError, ConfigurationError
+from subliminal.exceptions import ProviderError
+from ..subtitle import Subtitle, fix_line_ending, guess_matches, compute_score
+from ..video import Episode, Movie, SUBTITLE_EXTENSIONS
+from guessit import guess_file_info
+
+TIMEOUT = 10
+
+logger = logging.getLogger(__name__)
+
+server_url = 'http://legendas.tv'
+auth_error_msg_re = re.compile(u'.*Usuário ou senha inválidos.*')
+season_re = re.compile('.*? - (\d{1,2}).*?emporada')
+rating_info_re = re.compile('(\d*) downloads, nota (\d{0,2})')
+imdb_re = re.compile('t{0,2}(\d+)')
+subtitle_href_re = re.compile('/download/(\w+)/.+')
+
+mapping = dict(
+    id='id_filme',
+    name='dsc_nome',
+    name_br='dsc_nome_br',
+    year='dsc_data_lancamento',
+    type='tipo',
+    season='temporada',
+    imdb_id='id_imdb'
+)
+
+typemap = {
+    'M': 'movie',
+    'S': 'episode',
+    'C': 'episode'  # Considering C as episode. Probably C stands for Collections
+}
+
+
+def has_class(tag, expected_class):
+    while tag:
+        if not tag:
+            return False
+        if tag.has_attr('class') and expected_class in tag['class']:
+            return True
+        tag = tag.parent
+    return False
+
+
+def as_int(value, default=None):
+    return int(value) if value and value.isdigit() else default
+
+
+def extract(regex, value, group=1):
+    matcher = regex.search(value) if value else None
+    if matcher:
+        return matcher.group(group)
+
+
+def element_at(array, index):
+    return array[index] if index < len(array) else None
+
+
+def guess_info(name, type):
+    return guess_file_info(name + '.mkv', type=type)
+
+
+def select_best_name(names, subtitle):
+    if not names or len(names) == 0:
+        return
+
+    # Single name. Easy to choose
+    if len(names) == 1:
+        return names[0]
+
+    if not subtitle.video:
+        # Video is needed to guess the best subtitle to extract
+        raise ProviderError('More than one file to extract and no video defined')
+
+    selected_name = None
+    higher_score = 0
+    for name in names:
+        base_name = element_at(os.path.splitext(name), 0)
+        guess = guess_file_info(base_name + '.mkv', type=subtitle.type)
+        score = compute_score(guess_matches(subtitle.video, guess), subtitle.video)
+
+        if score > higher_score:
+            selected_name = name
+            higher_score = score
+
+    return selected_name
```

Comment:
Then don't use the download method. List subtitles and download them at the same time. One LegendasTvSubtitle == one .srt file in the rar/zip file.